### PR TITLE
Feat/comments

### PR DESCRIPTION
**Pull request by othniel-agera**
- Add comment
- Edit comment
- Get comments
- Delete comment

### CODE REVIEW

File: lib/comment.lib.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  const {
+    user_id, id, comment_text,
+  } = commentDetails;
+  try {
+    const comment = await fetchCommentOnQuestion({ user_id, id });
+    if (comment) {
+      comment.comment_text = comment_text;
+      await comment.save();
+      return comment;
+    }
```
Comment (kufre-okon) [inline lib/comment.lib.js:47]: Check if it's the user that made that comment is the one editing.

File: lib/comment.lib.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  } catch (error) {
+    throw new Error(`${error.message}`);
+  }
+};
+
+const destroyCommentOnQuestion = async (user_id, id) => {
+  try {
+    const comment = await QuestionComments.findOne({ where: { user_id, id } });
+    if (comment) return await comment.destroy(id);
+    throw new Error('Sorry no matching comment');
```
Comment (kufre-okon) [inline lib/comment.lib.js:58]: Check it the commenter that's attempting deletion.

File: lib/comment.lib.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+const editCommentOnAnswer = async (commentDetails) => {
+  const {
+    user_id, id, comment_text,
+  } = commentDetails;
+  try {
+    const comment = await fetchAnswerComment({ user_id, id });
+    if (comment) {
+      comment.comment_text = comment_text;
+      await comment.save();
```
Comment (kufre-okon) [inline lib/comment.lib.js:103]: Same as above, check that it's the right user performing this action.

File: lib/comment.lib.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  } catch (error) {
+    throw new Error(`${error.message}`);
+  }
+};
+
+const destroyCommentOnAnswer = async (user_id, id) => {
+  try {
+    const comment = await AnswerComments.findOne({ where: { user_id, id } });
+    if (comment) return await comment.destroy(id);
+    throw new Error('Sorry no matching comment');
```
Comment (kufre-okon) [inline lib/comment.lib.js:116]: Same as above, the it's the commenter attempting this action

**Review by kufre-okon — CHANGES_REQUESTED** [review]
Nothing much to address. Good job 🙏

File: lib/comment.lib.js (hunk 1 above)
Comment (othniel-agera) [reply lib/comment.lib.js:47]: On line 42 its actually checking if its the logged-in user that made the comment

File: lib/comment.lib.js (hunk 2 above)
Comment (othniel-agera) [reply lib/comment.lib.js:58]: On line 42 its actually checking if its the logged-in user that made the comment

File: lib/comment.lib.js (hunk 3 above)
Comment (othniel-agera) [reply lib/comment.lib.js:103]: On line 100 its actually checking if its the logged-in user that made the comment

File: lib/comment.lib.js (hunk 4 above)
Comment (othniel-agera) [reply lib/comment.lib.js:116]: It is the same as elsewhere, the logged-in user is checked to see if the made the comment before any of these actions are carried out.

File: lib/comment.lib.js (hunk 1 above)
Comment (KufreOkon) [reply lib/comment.lib.js:47]: Yeah, thanks, I overlooked the `user_id` passed to the query

File: lib/comment.lib.js (hunk 2 above)
Comment (KufreOkon) [reply lib/comment.lib.js:58]: Yeah, thanks, I overlooked the `user_id` passed to the query

File: lib/comment.lib.js (hunk 3 above)
Comment (KufreOkon) [reply lib/comment.lib.js:103]: Yeah, thanks, I overlooked the `user_id` passed to the query

File: lib/comment.lib.js (hunk 4 above)
Comment (KufreOkon) [reply lib/comment.lib.js:116]: Thanks

**Review by KufreOkon — APPROVED** [review]

**Pull request merged**